analyzers/rolaguard_base_analyzer/ABPDetector.py

```diff
@@ -0,0 +1,79 @@
+from collections import defaultdict
+import logging as log
+
+from utils.AlertGenerator import emit_alert
+from db.Models import DataCollector, Quarantine, AlertType
+
+
+
+class ABPDetector():
+    def __init__(self):
+        self.last_packet = defaultdict(lambda: {})
+
+    def __call__(self, packet, device_session, device, gateway, policy):
+        if (
+            device is None or
+            gateway is None or
+            packet.f_count is None
+        ): return # Can't be done anything without these data.
+
+        if device.is_otaa: return # It's already detected as OTAA, nothing to do.
+
+        # Used to identify a "connection" to check. Here for connection we are 
+        # talking about the flow of packets between a gateway and a device in
+        # in the context of a stablished device_session.
+        lpacket_uid = (device.id, gateway.id)
+
+        if ( # This indicates that the device is OTAA
+            packet.m_type in ["JoinRequest", "JoinAccept"] or
+            (
+                packet.dev_addr is not None and
+                lpacket_uid in self.last_packet and
+                self.last_packet[lpacket_uid]["dev_addr"] != packet.dev_addr
+            )
+        ):
+            res_comment = "The device has sent a join request"
+            issue_solved = Quarantine.remove_from_quarantine(
+                "LAF-006",
+                device_id = device.id,
+                device_session_id = device_session.id,
```

Comment:
When resolving this kind of issues I think that it may be possible that the device_session here is different from the one that was set when the issue was created (due to line 32). If I am right, then leaving the device session as None (here and in line 66) should be enough for this to work.

Reply:
You are right! Done